### PR TITLE
[M] Fixed an issue with shared pools not being deprioritized

### PR DESCRIPTION
**Pull request by Ceiu**
- Fixed an issue with the hasSharedAncestor property not being
  serialized, preventing the rules from deprioritizing shared
  pools during pool selection

### CODE REVIEW

**Comment by Ceiu** [comment]
An alternative to this fix would be changing the rules to use isCreatedByShare rather than exposing hasSharedAncestor to the JSON. Some guidance from @awood or @vritant would be welcome here.

**Comment by awood** [comment]
> An alternative to this fix would be changing the rules to use isCreatedByShare rather than exposing hasSharedAncestor to the JSON. Some guidance from @awood or @vritant would be welcome here.

I would rather have the fix in `isCreatedByShare` to be honest.

File: server/src/main/resources/rules/rules.js [hunk 1]
```diff
@@ -326,36 +326,43 @@ function get_mock_ent_for_pool(pool, consumer) {
...
     }
+
     // better still if host_specific
     if (pool.getAttribute(REQUIRES_HOST_ATTRIBUTE) !== null) {
         priority += 150;
     }
 
     // Decrease the priority of shared pools slightly so that non-shared pools will get consumed first.
-    if (pool.hasSharedAncestor) {
+    if (pool.createdByShare || pool.hasSharedAncestor) {
```
Comment (vritant) [inline server/src/main/resources/rules/rules.js]: using hasSharedAncestor should be enough, so I would argue we should use **only** that, cause this use was the sole purpose of this field.

Comment (Ceiu) [reply server/src/main/resources/rules/rules.js]: Gotta say, I'm not a fan of those two things. For the sake of this PR, I'll revert it, but going forward we need to pair that down a bit. Having two columns that describe the same thing with only slight differences is going to lead to problems down the road, and has already burned a bit of time trying to understand intent and significant difference.

Comment (awood) [reply server/src/main/resources/rules/rules.js]: I think we need to preserve the distinction but decide on better names.  As I understand it:

* createdByShare is for a pool that is at the top of the sharing tree.
* hasSharedAncestor is true if any pool in a pool's lineage or if the pool itself is shared.

Perhaps something like `isShareRoot` and `isConsideredShare`?

**Pull request merged**